src/muse/carbon_budget.py

```diff
@@ -138,72 +125,14 @@ def fitting(
         threshold,  # type: ignore
     )
 
-    if refine_price and new_market is not None:
-        new_price = refine_new_price(
-            new_market,
-            carbon_price,
-            carbon_budget,
-            sample_prices,
-            new_price,
-            commodities,
-            price_too_high_threshold,
-        )
+    # Cap price between 0.01 and price_too_high_threshold
+    if refine_price:
+        new_price = min(new_price, price_too_high_threshold)
```

Comment:
Why is there so much difference between the original refine price function and this one, which just one line?

Reply:
There was some additional logic in the original function that was looking at the growth rate of the carbon price, but it doesn't seem to be properly implemented and it's not clear what the intended behaviour was. There might be something to think about here, like putting an upper limit on the rate of price increase/decrease, but the documentation only talked about capping the price based on `price_too_high_threshold`, so this is what I went for